.github/workflows/codeql.yml

```diff
@@ -0,0 +1,104 @@
+        # Use only 'javascript' to analyze code written in JavaScript, TypeScript or both
+        # Learn more about CodeQL language support at https://aka.ms/codeql-docs/language-support
+
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v3
+
+      # Initializes the CodeQL tools for scanning.
+      - name: Initialize CodeQL
+        uses: github/codeql-action/init@v2
+        with:
+          languages: ${{ matrix.language }}
+          # If you wish to specify custom queries, you can do so here or in a config file.
+          # By default, queries listed here will override any specified in a config file.
+          # Prefix the list here with "+" to use these queries and those in the config file.
+
+          # Details on CodeQL's query packs refer to : https://docs.github.com/en/code-security/code-scanning/automatically-scanning-your-code-for-vulnerabilities-and-errors/configuring-code-scanning#using-queries-in-ql-packs
+          # queries: security-extended,security-and-quality
+
+
+      # Autobuild attempts to build any compiled languages  (C/C++, C#, Go, or Java).
+      # If this step fails, then you should remove it and run the build manually (see below)
+      - name: Autobuild
+        uses: github/codeql-action/autobuild@v2
+
+      # ℹ️ Command-line programs to run using the OS shell.
+      # 📚 See https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#jobsjob_idstepsrun
+
+      #   If the Autobuild fails above, remove it and uncomment the following three lines.
+      #   modify them (or add more) to build your code if your project, please refer to the EXAMPLE below for guidance.
+
+      # - run: |
+      #   echo "Run, Build Application using script"
+      #   ./location_of_script_within_repo/buildscript.sh
+
+      - name: Perform CodeQL Analysis
+        uses: github/codeql-action/analyze@v2
+        with:
+          category: "/language:${{matrix.language}}"
+
+  # .Net is run as a separate job as the integration tests take far too long to build
+  analyze_dotnet:
+    name: Analyze .Net
+    runs-on: ubuntu-latest
+    permissions:
+      actions: read
+      contents: read
+      security-events: write
+
+    strategy:
+      fail-fast: false
+      matrix:
+        language: [ 'csharp' ]
+
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v3
+
+      # Initializes the CodeQL tools for scanning.
+      - name: Initialize CodeQL
+        uses: github/codeql-action/init@v2
+        with:
+          languages: ${{ matrix.language }}
+
+      - run: |
+          echo "Build .NET"
+          dotnet build gremlin-dotnet/src/Gremlin.Net;
+          dotnet build gremlin-dotnet/src/Gremlin.Net.Template
+          dotnet build gremlin-dotnet/test/Gremlin.Net.UnitTest
+          dotnet build gremlin-dotnet/test/Gremlin.Net.Benchmarks
+#          Excluded due to extremely long (~45m) build time in github actions
```

Comment:
Wow, that's really extreme. Would be interesting to know of course what's taking so long here, but it's also not that important, especially for this PR. I also don't think that we really need to perform the scanning on test projects as we don't ship them to users.

Reply:
Yeah I was pretty surprised by that result but in my tests it was very consistent and reproducible. I'm sure a build system could be found which doesn't run into whatever bottleneck this is hitting. It doesn't seem like a much of a priority at this time though.